test/unit/test_randomai.gd

```diff
@@ -17,31 +14,25 @@ var player1 : LocalGame.Player
 var player2 : LocalGame.Player
 var ai1 : AIPlayer
 var ai2 : AIPlayer
-var ai_policy
 
-func game_setup(policy = AIPolicyRules.new()):
-	ai_policy = policy
+func game_setup(policy_type = AIPolicyRules):
 	game_logic = LocalGame.new()
 	var seed_value = randi()
 	game_logic.initialize_game(default_deck, default_deck, "p1", "p2", Enums.PlayerId.PlayerId_Player, seed_value)
 	game_logic.draw_starting_hands_and_begin()
 	game_logic.get_latest_events()
 	player1 = game_logic.player
 	player2 = game_logic.opponent
-	ai1 = AIPlayer.new()
-	ai1.ai_policy.free()
-	ai1.set_ai_policy(ai_policy)
-	ai1.game_player = player1
-	ai2 = AIPlayer.new()
-	ai2.ai_policy.free()
-	ai2.set_ai_policy(ai_policy)
-	ai2.game_player = player2
+	ai1 = AIPlayer.new(game_logic, player1, policy_type.new())
+	ai2 = AIPlayer.new(game_logic, player2, policy_type.new())
 
 func game_teardown():
+	# TODO: Move this logic into the real game so that it doesn't memory leak
```

Comment:
As you can probably tell, I've been a bit of a mess with godot memory management.
game wrapper should probably be freed at some point in the main scene? and have teardowns to free localgame/remotegame.
Or maybe made into a node that attaches to the scene so it gets freed automatically?

Reply:
I'm not 100% on it myself but the unit tests do a good job of complaining when it's not being handled fully. The main thing here is that as far as I can tell, only `RefCounted` and its descendents (of which `Resource` is one) get automatically garbage-collected, and everything else has to be manually `free`d or `queue_free`d.

The TODO is mostly the idea that if you have to make multiple `free()` calls manually in the tests, that probably means that there's a `teardown()` belonging somewhere else -- maybe in `game_wrapper`? -- that should hold the top-level piece of this logic and recursively `free()` its children (like any Player objects, AI or otherwise).